src/miexp/bfuncs.py

```diff
@@ -151,3 +151,47 @@ def get_percent_positive(self) -> float:
             float: The mean value of the labels, representing the percentage of positive labels.
         """
         return self.labels.type(torch.float).mean().item()
+
+
+class MultiComponentDNFDataset(Dataset):
+    """Generate a dataset for the multi-component DNF boolean function."""
+
+    def __init__(self, N: int, comps: Tensor, num_samples: int = 10000) -> None:  # noqa: N803
+        """Initialize dataset for the multi-component DNF boolean function.
+
+        Args:
+            N (int): Size of the input boolean string
+            comps (Tensor): The DNF components (boolean valued) (Shape: (num_components, N))
```

Comment:
I don't think the component indicator vectors of a DNF are boolean valued. Aren't they ternary? One component either requires a bit to be 1, 0, or doesn't care